hazelcast-sql/src/main/java/com/hazelcast/sql/impl/schema/MappingField.java

```diff
@@ -89,13 +89,21 @@ public MappingField setType(QueryDataType type) {
     }
 
     /**
-     * The external name of a field. For example, in case of IMap or Kafka,
+     * The external name of this field. For example, in case of IMap or Kafka,
      * it always starts with `__key` or `this`.
      */
     public String externalName() {
         return (String) properties.get(EXTERNAL_NAME);
     }
 
+    /**
+     * The external name of this field without `__key` or `this` prefix.
+     */
+    public String plainExternalName() {
```

Comment:
this method might be ill-defined for connectors that do not distinguish key/value. Also parsing external identifier (to find dot) is risky.
`QueryPath.create` has a more robust way to do this (but still not perfect).

Reply:
It is just introduced to access the plain name more easily is tests. If you elaborate on the _risks_, I can improve the definition/implementation.

Reply:
if it is meant only for tests mark it as such.
Example problems: escaping ("my\.field"), other field names with dots (maybe not supported currently, but why risk hard to find errors in the future?)
I also think that "it always starts with `__key` or `this`." might not be true for JDBC or Mongo. External name can also be null for them.

Reply:
I decided to use `QueryPath.create` for parsing and move the method to somewhere else since the interpretation of _external name_ is up to the connector.